File: dailyQuestion/2021/2021-03/03-20/python/solution.py
class Solution:
    def evalRPN(self, tokens: [str]) -> int:
        stack = []
        for i in range(len(tokens)) :
            tmp = tokens[i]
            if tmp == "+" :
                num1 = stack.pop()
                num2 = stack.pop()
                stack.append(num2 + num1)
            elif tmp == "-" :
                num1 = stack.pop()
                num2 = stack.pop()
                stack.append(num2 - num1)
            elif tmp == "*" :
                num1 = stack.pop()
                num2 = stack.pop()
                stack.append(num2 * num1)
            elif tmp == "/" :
                num1 = stack.pop()
                num2 = stack.pop()
                stack.append(int(num2 / num1))
            else :
                stack.append(int(tmp))
        return stack[0]

if __name__ == "__main__" :
    tokens = ["10","6","9","3","+","-11","*","/","*","17","+","5","+"]
    s = Solution()
    res = s.evalRPN(tokens)
    print(res)

File: dailyQuestion/2021/2021-03/03-20/golang/solution.go
package main

import (
	"fmt"
	"strconv"
)

func main() {
	tokens := []string{"10", "6", "9", "3", "+", "-11", "*", "/", "*", "17", "+", "5", "+"}
	res := evalRPN(tokens)
	fmt.Println(res)
}

func evalRPN(tokens []string) int {
	stack := []int{}
	for i := 0; i < len(tokens); i++ {
		tmp := tokens[i]
		switch tmp {
		case "+":
			num1 := stack[len(stack)-1]
			num2 := stack[len(stack)-2]
			stack = stack[:len(stack)-2]
			stack = append(stack, num2+num1)
		case "-":
			num1 := stack[len(stack)-1]
			num2 := stack[len(stack)-2]
			stack = stack[:len(stack)-2]
			stack = append(stack, num2-num1)
		case "*":
			num1 := stack[len(stack)-1]
			num2 := stack[len(stack)-2]
			stack = stack[:len(stack)-2]
			stack = append(stack, num2*num1)
		case "/":
			num1 := stack[len(stack)-1]
			num2 := stack[len(stack)-2]
			stack = stack[:len(stack)-2]
			stack = append(stack, num2/num1)
		default:
			num, _ := strconv.Atoi(tmp)
			stack = append(stack, num)
		}
	}
	return stack[0]
}
